src/mcts/stoppers/alphazero.cc

```diff
@@ -36,33 +36,40 @@ class AlphazeroTimeManager : public TimeManager {
   AlphazeroTimeManager(int64_t move_overhead, const OptionsDict& params)
       : move_overhead_(move_overhead),
         alphazerotimepct_(
-            params.GetOrDefault<float>("alphazero-time-pct", 12.0f)) {
+            params.GetOrDefault<float>("alphazero-time-pct", 12.0f)),
+        alphazeroinctimepct_(
+            params.GetOrDefault<float>("alphazero-inctime-pct", 80.0f)) {
     if (alphazerotimepct_ < 0.0f || alphazerotimepct_ > 100.0f)
       throw Exception("alphazero-time-pct value to be in range [0.0, 100.0]");
+    if (alphazeroinctimepct_ < 0.0f || alphazeroinctimepct_ > 100.0f)
+      throw Exception("alphazero-inctime-pct value to be in range [0.0, 100.0]");
   }
   std::unique_ptr<SearchStopper> GetStopper(const GoParams& params,
                                             const NodeTree& tree) override;
 
  private:
   const int64_t move_overhead_;
   const float alphazerotimepct_;
+  const float alphazeroinctimepct_;
 };
 
 std::unique_ptr<SearchStopper> AlphazeroTimeManager::GetStopper(
     const GoParams& params, const NodeTree& tree) {
   const Position& position = tree.HeadPosition();
   const bool is_black = position.IsBlackToMove();
   const std::optional<int64_t>& time = (is_black ? params.btime : params.wtime);
+  const std::optional<int64_t>& time_inc = (is_black ? params.binc : params.winc);
   // If no time limit is given, don't stop on this condition.
   if (params.infinite || params.ponder || !time) return nullptr;
 
   auto total_moves_time = *time - move_overhead_;
 
-  float this_move_time = total_moves_time * (alphazerotimepct_ / 100.0f);
+  float this_move_time = (total_moves_time - *time_inc) * (alphazerotimepct_ / 100.0f)
```

Comment:
I'm not sure what is the best way to handle this. Subtracting `time_inc` may result in negative `this_move_time`. My initial thinking was that just using `total_moves_time` like in the first iteration is OK, and it is safe as we guard against exceeding the time limit.

Reply:
I noticed that `wtime` is always main+increment time added (at least in Cutechess). So `this_move_time` should never become negative.

Reply:
At least make it `std::max(0, total_moves_time - *time_inc)`. I'm not sure all uci hosts do it this way.